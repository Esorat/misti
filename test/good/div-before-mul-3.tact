fun test(): Int {
  let a: Int = 10 / 3; // Division operation
  let c: Int = 2;
  let taintedWithA1: Int = 2 - a;
  let taintedWithA2: Int = 55 + taintedWithA1;
  let result: Int = taintedWithA2 * c; // Error: Using a variable tainted with division
  return result;
}

