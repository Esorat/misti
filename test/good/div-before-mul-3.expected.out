test/good/div-before-mul-3.tact:2:16:
  1 | fun test(): Int {
> 2 |   let a: Int = 10 / 3; // Division operation
                     ^
  3 |   let c: Int = 2;
Divide Before Multiply
Help: Consider rearranging the operations: division should follow multiplication
See: https://nowarp.github.io/docs/misti/docs/detectors/DivideBeforeMultiply

test/good/div-before-mul-3.tact:2:3:
  1 | fun test(): Int {
> 2 |   let a: Int = 10 / 3; // Division operation
        ^
  3 |   let c: Int = 2;
Read-only variable
Help: Consider creating a constant instead
See: https://nowarp.github.io/docs/misti/docs/detectors/ReadOnlyVariables

test/good/div-before-mul-3.tact:3:3:
  2 |   let a: Int = 10 / 3; // Division operation
> 3 |   let c: Int = 2;
        ^
  4 |   let taintedWithA1: Int = 2 - a;
Read-only variable
Help: Consider creating a constant instead
See: https://nowarp.github.io/docs/misti/docs/detectors/ReadOnlyVariables

test/good/div-before-mul-3.tact:4:3:
  3 |   let c: Int = 2;
> 4 |   let taintedWithA1: Int = 2 - a;
        ^
  5 |   let taintedWithA2: Int = 55 + taintedWithA1;
Read-only variable
Help: Consider creating a constant instead
See: https://nowarp.github.io/docs/misti/docs/detectors/ReadOnlyVariables

test/good/div-before-mul-3.tact:5:3:
  4 |   let taintedWithA1: Int = 2 - a;
> 5 |   let taintedWithA2: Int = 55 + taintedWithA1;
        ^
  6 |   let result: Int = taintedWithA2 * c; // Error: Using a variable tainted with division
Read-only variable
Help: Consider creating a constant instead
See: https://nowarp.github.io/docs/misti/docs/detectors/ReadOnlyVariables

test/good/div-before-mul-3.tact:6:3:
  5 |   let taintedWithA2: Int = 55 + taintedWithA1;
> 6 |   let result: Int = taintedWithA2 * c; // Error: Using a variable tainted with division
        ^
  7 |   return result;
Read-only variable
Help: Consider creating a constant instead
See: https://nowarp.github.io/docs/misti/docs/detectors/ReadOnlyVariables